File: frogment/src/main/java/ch/grze/frogment/activity/ActivityCallbacks.java
package ch.grze.frogment.activity;

import android.app.Activity;
import android.os.Bundle;

import ch.grze.frogment.core.Core;
import ch.grze.frogment.core.callbacks.AbstractActivityLifecycleCallbacks;

public class ActivityCallbacks extends AbstractActivityLifecycleCallbacks {
    public ActivityCallbacks(Core core) {
        super(core);
    }

    @Override
    public void onActivityCreated(Activity activity, Bundle bundle) {
        super.onActivityCreated(activity, bundle);

        try {
            final FrogmentActivityInterface frogmentActivity = getTypedActivity(activity);
            frogmentActivity.getFrogmentActivityComponent().onActivityCreated(bundle);
        } catch (ClassCastException ignored) {}

        try {
            final StateAwareFrogmentActivityInterface stateAwareFrogmentActivity = getTypedActivity(activity);
            stateAwareFrogmentActivity.getStateAwareFrogmentActivityComponent().onActivityCreated(bundle);
        } catch (ClassCastException ignored) {}
    }

    @Override
    public void onActivityStarted(Activity activity) {
        super.onActivityStarted(activity);

        try {
            final StateAwareFrogmentActivityInterface stateAwareFrogmentActivity = getTypedActivity(activity);
            stateAwareFrogmentActivity.getStateAwareFrogmentActivityComponent().onActivityStarted();
        } catch (ClassCastException ignored) {}
    }

    @Override
    public void onActivitySaveInstanceState(Activity activity, Bundle bundle) {
        super.onActivitySaveInstanceState(activity, bundle);

        try {
            final FrogmentActivityInterface frogmentActivity = getTypedActivity(activity);
            frogmentActivity.getFrogmentActivityComponent().onActivitySaveInstanceState(bundle);
        } catch (ClassCastException ignored) {}

        try {
            final StateAwareFrogmentActivityInterface stateAwareFrogmentActivity = getTypedActivity(activity);
            stateAwareFrogmentActivity.getStateAwareFrogmentActivityComponent().onActivitySaveInstanceState(bundle);
        } catch (ClassCastException ignored) {}
    }
}

File: frogment/src/main/java/ch/grze/frogment/activity/ActivityCallbacks.kt
package ch.grze.frogment.activity

import android.app.Activity
import android.os.Bundle
import ch.grze.frogment.core.Core
import ch.grze.frogment.core.callbacks.AbstractActivityLifecycleCallbacks

class ActivityCallbacks(core: Core) : AbstractActivityLifecycleCallbacks(core) {
    override fun onActivityCreated(activity: Activity, bundle: Bundle?) {
        super.onActivityCreated(activity, bundle)

        (activity as? FrogmentActivityInterface)
            ?.frogmentActivityComponent
            ?.onActivityCreated(bundle)

        (activity as? StateAwareFrogmentActivityInterface<*>)
            ?.stateAwareFrogmentActivityComponent
            ?.onActivityCreated(bundle)
    }

    override fun onActivityStarted(activity: Activity) {
        super.onActivityStarted(activity)

        (activity as? StateAwareFrogmentActivityInterface<*>)
            ?.stateAwareFrogmentActivityComponent
            ?.onActivityStarted()
    }

    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {
        super.onActivitySaveInstanceState(activity, outState)

        (activity as? FrogmentActivityInterface)
            ?.frogmentActivityComponent
            ?.onActivitySaveInstanceState(outState)

        (activity as? StateAwareFrogmentActivityInterface<*>)
            ?.stateAwareFrogmentActivityComponent
            ?.onActivitySaveInstanceState(outState)
    }
}